src/main/java/net/starlark/java/syntax/StarlarkStringInterner.java

```diff
@@ -0,0 +1,21 @@
+package net.starlark.java.syntax;
+
+import com.google.common.collect.Interner;
+import com.google.common.collect.Interners;
+
+/**
+ * Common place to intern strings in Starlark interpreter.
+ *
+ * <p>Interned strings are much faster to lookup, which is important, for example, when evaluating
+ * expression {@code foo.bar}.
+ */
+public class StarlarkStringInterner {
+  private StarlarkStringInterner() {}
+
+  private static final Interner<String> INTERNER = Interners.newWeakInterner();
+
+  /** Weak intern the string. */
+  public static String intern(String string) {
+    return INTERNER.intern(string);
+  }
+}
```

Comment:
This is going to be the third string interner in Starlark; we already have Lexer.identInterner and Parser.stringInterner.

I would much prefer to coalesce these interners into one somehow, and if that's not possible, at least explain which interner should be used in which circumstance and why.